tests/src/actionContainers/ActionProxyContainerTests.scala

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package actionContainers
+
+import org.junit.runner.RunWith
+import org.scalatest.Finders
+import org.scalatest.junit.JUnitRunner
+
+import ActionContainer.withContainer
+import spray.json.JsNull
+import spray.json.JsNumber
+import spray.json.JsObject
+import spray.json.JsString
+
+@RunWith(classOf[JUnitRunner])
+class ActionProxyContainerTests extends ActionProxyContainerTestUtils {
+
+    def withPythonContainer(code: ActionContainer => Unit) = withContainer("whisk/dockerskeleton")(code)
+
+    behavior of "whisk/dockerskeleton"
+
+    it should "run sample without init" in {
+        val (out, err) = withPythonContainer { c =>
+            val (runCode, out) = c.run(JsNull)
+            runCode should be(200)
+            out should be(Some(JsObject("msg" -> JsString("Hello from bash script!"))))
+        }
+
+        checkStreams(out, err, {
+            case (o, _) => o should include("Actions may log to stdout")
+        })
+    }
+
+    it should "run sample with init that does nothing" in {
```

Comment:
There are two points here (and in the related comment above):

- should the image have a default, built-in action? I would have thought not, but I can see that perhaps the goal is to make errors user-friendly when someone extends the image but, e.g., puts their executable at the wrong location? In that case, the message should perhaps be "this is the default action", or something in that vein? (see for instance the Apache 2 default page that includes instructions on how to make it go away)

- should POSTing `null` to `/init` be valid/invalid/undefined? I don't see why we would want a test that makes it valid. I see that the comments in the proxy say this will be detected later, but falling back to the default action will just mask the error in a strange way IMHO.